### PR TITLE
fix(YouTube - Remove background playback restrictions): Enable for Short as well

### PR DESCRIPTION
**Pull request by zainarbani**
Integrations for: https://github.com/ReVanced/revanced-patches/pull/3671

### CODE REVIEW

File: app/src/main/java/app/revanced/integrations/youtube/patches/BackgroundPlaybackPatch.java [hunk 1]
```diff
@@ -20,16 +21,11 @@ public static boolean playbackIsNotShort() {
...
-            return true; // Definitely is not a Short.
+        if (VideoInformation.lastVideoIdIsShort()) {
+            return !Settings.DISABLE_BACKGROUND_SHORTS.get();
         }
 
-        // Might be a Short, or might be a prior regular video on screen again after a Short was closed.
-        // This incorrectly prevents PIP if player is in WATCH_WHILE_MINIMIZED after closing a Short,
-        // But there's no way around this unless an additional hook is added to definitively detect
-        // the Shorts player is on screen. This use case is unusual anyways so it's not a huge concern.
-        return !PlayerType.getCurrent().isNoneHiddenOrMinimized();
```
Comment (zainarbani) [inline app/src/main/java/app/revanced/integrations/youtube/patches/BackgroundPlaybackPatch.java:32]: @LisoUseInAIKyrios I suppose this line is no longer needed?

Comment (zainarbani) [reply app/src/main/java/app/revanced/integrations/youtube/patches/BackgroundPlaybackPatch.java:32]: The issue apparently still there

File: app/src/main/java/app/revanced/integrations/youtube/patches/BackgroundPlaybackPatch.java [hunk 2]
```diff
@@ -21,11 +21,21 @@ public static boolean isBackgroundPlaybackAllowed() {
...
+            return true; // Definitely is not a Short.
         }
 
-        return true;
+        // TODO: Add better hook.
+        // Might be a Shorts, or might be a prior regular video on screen again after a Shorts was closed.
+        // This incorrectly prevents PIP if player is in WATCH_WHILE_MINIMIZED after closing a Shorts,
+        // But there's no way around this unless an additional hook is added to definitively detect
+        // the Shorts player is on screen. This use case is unusual anyways so it's not a huge concern.
+        if (!PlayerType.getCurrent().isNoneHiddenOrMinimized()) {
```
Comment (zainarbani) [inline app/src/main/java/app/revanced/integrations/youtube/patches/BackgroundPlaybackPatch.java:33]: @LisoUseInAIKyrios I think `REEL_WATCH_FRAGMENT_INIT_PLAYBACK.contains(tag)` can be utilized here.

Comment (zainarbani) [reply app/src/main/java/app/revanced/integrations/youtube/patches/BackgroundPlaybackPatch.java:33]: Understood

**Comment by oSumAtrIX** [comment]
With PR https://github.com/ReVanced/revanced-patches/pull/3462, ReVanced Integrations has been merged into ReVanced Patches. This PR must be reopened in the [ReVanced Patches](https://github.com/ReVanced/revanced-patches) repository and adjusted to the new build system. ReVanced Integration is now understood as "extensions" according to the ReVanced Patcher API. As extensions are now on a per-patch basis, this PR may be subject to a separate extension. If not, the PR can change the shared extension in [ReVanced Patches](https://github.com/ReVanced/revanced-patches).

Thanks, for the PR.

**Pull request closed without merging**